### PR TITLE
Perform validation on files in REST layer

### PR DESCRIPTION
**Pull request by zachmullen**
@brianhelba PTAL, my goal here is to ensure that the REST layer gives 400 errors with reasonable messages when model create/update fails validation. I have no idea where is the best place to put this validation logic.

### CODE REVIEW

**Comment by zachmullen** [comment]
Nobody seems to have a great answer for where to do validation... if you know of a better way, I'd love to hear it.

File: dkc/core/rest/utils.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from django.core.exceptions import ValidationError as DjangoValidationError
+from rest_framework import serializers
+
+
+class FullCleanModelSerializer(serializers.ModelSerializer):
+    def validate(self, data):
+        if self.partial:  # partial update
+            for k, v in data.items():
+                setattr(self.instance, k, v)
```
Comment (brianhelba) [inline dkc/core/rest/utils.py:9]: We're mutating the only copy of `self.instance` here, which means that validation (even if unsuccessful) has side effects on the existing model. If a request doesn't terminate immediately after this, we'll end up with an invalid model, which could even be a problem for debugging.

Comment (zachmullen) [reply dkc/core/rest/utils.py]: > If a request doesn't terminate immediately after this, we'll end up with an invalid model, which could even be a problem for debugging.

Do you mean to say that a model instance object may be shared across different requests...?

Comment (zachmullen) [reply dkc/core/rest/utils.py]: Ignore deleted comment, I misread your last sentence.

**Review by brianhelba — APPROVED** [review]
Seems reasonable for now. I hope that as our business logic becomes more complex, it will be more obvious where the most natural place for it is.

**Pull request merged**